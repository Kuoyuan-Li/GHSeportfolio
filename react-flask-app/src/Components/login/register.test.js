import React from 'react'
import Register from './register'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer';

describe ('login page test', () => {
    

    const historyMock = { push: jest.fn() }
    const onMock = jest.fn();

    let wrapper;

    beforeAll(() => {
        global.fetch = jest.fn();
    });

    beforeEach(() => {
        wrapper = shallow(<Register history = {historyMock} onSubmit={onMock}/>); 
    });

    test('redirect to index page when click : Back to Index Page', () => {
        wrapper.find('button').at('1').simulate('click');       
        expect(historyMock.push.mock.calls[0]).toEqual(['/']);
    });

    test('Input incomplete register Info, miss password 2' , ()=>{
        const form = wrapper.find('form.registerForm');
        form.simulate('submit', {
            preventDefault: () => {},
            target: {
                elements: {
                    username: { value: '1' },
                    email : { value : '1'},
                    password: { value: '1' },
                    password2: { value: '' },
                } 
            }
        });
        expect(wrapper.state('message')).toEqual('Please enter all required information');      
    });

    test('Input incomplete register Info, miss password ' , ()=>{
        const form = wrapper.find('form.registerForm');
        form.simulate('submit', {
            preventDefault: () => {},
            target: {
                elements: {
                    username: { value: '1' },
                    email : { value : '1'},
                    password: { value: '' },
                    password2: { value: '1' },
                } 
            }
        });
        expect(wrapper.state('message')).toEqual('Please enter all required information');      
    });

    test('Input incomplete register Info, miss email' , ()=>{
        const form = wrapper.find('form.registerForm');
        form.simulate('submit', {
            preventDefault: () => {},
            target: {
                elements: {
                    username: { value: '1' },
                    email : { value : ''},
                    password: { value: '1' },
                    password2: { value: '1' },
                } 
            }
        });
        expect(wrapper.state('message')).toEqual('Please enter all required information');      
    });

    test('Input incomplete register Info, miss username' , ()=>{
        const form = wrapper.find('form.registerForm');
        form.simulate('submit', {
            preventDefault: () => {},
            target: {
                elements: {
                    username: { value: '' },
                    email : { value : '1'},
                    password: { value: '1' },
                    password2: { value: '1' },
                } 
            }
        });
        expect(wrapper.state('message')).toEqual('Please enter all required information');      
    });
   
    test('renders correctly', () => {
        const tree = renderer.create(<Register />).toJSON();
        expect(tree).toMatchSnapshot();
      });
});