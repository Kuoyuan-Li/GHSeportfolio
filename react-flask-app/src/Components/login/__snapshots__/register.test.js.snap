// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page test renders correctly 1`] = `
<body
  id="login"
>
  <div
    className="container"
  >
    <div
      class="row"
    >
      <form
        className="registerForm"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div />
        <div
          className="form"
        >
          <div
            className="form-group"
          >
            <input
              name="username"
              onChange={[Function]}
              placeholder="Type your user name"
              type="username"
              value=""
            />
          </div>
          <div
            className="form-group"
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <input
              name="email"
              onChange={[Function]}
              placeholder="Type your email address"
              type="email"
              value=""
            />
            <button
              class="Button"
              onClick={[Function]}
            >
              Send verification code
            </button>
          </div>
          <div
            className="form-group"
          >
            <input
              name="userCaptcha"
              onChange={[Function]}
              placeholder="Type the verification code sent to your email"
              type="captcha"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              name="password"
              onChange={[Function]}
              placeholder="Type your password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              name="password2"
              onChange={[Function]}
              placeholder="Re-enter password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <button
              class="button button2"
              type="submit"
            >
              Create account
            </button>
          </div>
        </div>
      </form>
    </div>
    <button
      class="linkButton"
      onClick={[Function]}
    >
      Back to Index Page
    </button>
  </div>
</body>
`;
